### DISCUSSION TITLE
Route with ext (.html)

### DISCUSSION THREAD

**Discussion by nautonk** [adonisjs/core#4031] [Help]
Hi adonisjs,

I'm trying to achieve route with suffix .html, my code is :
`Route.get(`/detail/:id/:slug.html`,'Frontend/PostsController.getDetail').as('posts.getDetail')`

but i've got an error.

in Laravel i can achive this by using :
`Route::get('/detail/{id}/{slug}.html', [Frontend\PostsController::class, 'getDetail'])`

How to do this in adonisjs ?

Thank you so much

**Comment by tomgobich** [comment]
What error are you getting? I tested it out in one of my projects, see image below, and it seems to be working okay.

<img width="726" alt="Screen Shot 2023-01-06 at 4 23 17 PM" src="https://user-images.githubusercontent.com/19398726/211102620-9fa576c3-da48-44f3-808c-01f08b060fcb.png">

**Reply by nautonk** [reply]
> What error are you getting? I tested it out in one of my projects, see image below, and it seems to be working okay.
> 
> <img alt="Screen Shot 2023-01-06 at 4 23 17 PM" width="726" src="https://user-images.githubusercontent.com/19398726/211102620-9fa576c3-da48-44f3-808c-01f08b060fcb.png">

It works if i directly access to the url, but if i render through edge template it show error.
![Screenshot 2023-01-07 at 17 08 23](https://user-images.githubusercontent.com/3839865/211145078-fdc66bfb-7083-4ee3-bed6-09f31ad0bcad.jpg)

Thank you for helping me

**Reply by tomgobich** [reply]
Hmm, yeah it looks like the route generator considers everything between `/` and starting with `:` as the param name whereas the param within the `HttpContext` is using `@poppins/matchit`, which seems to get the param name correct.

Route Generator Params: https://github.com/adonisjs/http-server/blob/develop/src/Router/LookupStore.ts#L98-L126 
HttpContext Params: https://github.com/adonisjs/http-server/blob/develop/src/HttpContext/index.ts#L226-L229

This isn't an ideal solution, since you'd have to manually add the `.html` suffix, but for now, if you use either the param name `slug.html` when generating your route or use the array definition everything appears to be working okay for this. Then, since the `.html` portion is getting included in the param name, it's excluded in the generated URL, so you'll need to suffix the link with `.html`.

```html
<a href="{{ route('tester', { 'slug.html': x }) }}.html">Naming slug.html</a>
<a href="{{ route('tester', [x]) }}.html">Using array definition. This expects the param values as an array in the order they're defined in the route definition</a>
```

Edit:
Submitted issue within the Http-Server repo: https://github.com/adonisjs/http-server/issues/67

**Reply by nautonk** [reply]
> Hmm, yeah it looks like the route generator considers everything between `/` and starting with `:` as the param name whereas the param within the `HttpContext` is using `@poppins/matchit`, which seems to get the param name correct.
> 
> Route Generator Params: https://github.com/adonisjs/http-server/blob/develop/src/Router/LookupStore.ts#L98-L126 HttpContext Params: https://github.com/adonisjs/http-server/blob/develop/src/HttpContext/index.ts#L226-L229
> 
> This isn't an ideal solution, since you'd have to manually add the `.html` suffix, but for now, if you use either the param name `slug.html` when generating your route or use the array definition everything appears to be working okay for this. Then, since the `.html` portion is getting included in the param name, it's excluded in the generated URL, so you'll need to suffix the link with `.html`.
> 
> ```
> <a href="{{ route('tester', { 'slug.html': x }) }}.html">Naming slug.html</a>
> <a href="{{ route('tester', [x]) }}.html">Using array definition. This expects the param values as an array in the order they're defined in the route definition</a>
> ```
> 
> Edit: Submitted issue within the Http-Server repo: [adonisjs/http-server#67](https://github.com/adonisjs/http-server/issues/67)

Noted, at this time i will use your suggestion fix.

Thank you so much for your help bro. I really appreciate it, God bless you

**Reply by tomgobich** [reply]
Anytime!
